TombEngine/Scripting/Internal/TEN/Misc/Miscellaneous.cpp

```diff
@@ -393,6 +396,26 @@ namespace Misc
 		TENLog(message, level, LogConfig::All, USE_IF_HAVE(bool, allowSpam, false));
 	}
 
+	/// Get the projected 2D screen position of a 3D world position. Returns nil if the 3D position is behind the camera view.
+	//@tparam Vec3 pos3D 3D world position.
+	//@return Vec2 pos2D Projected 2D screen position. NOTE: Screen space resolution is 100x100.
+	//@usage 
+	//
+	// Example: Display a string at the player's position.
+	// local string = DisplayString('Example', 0, 0, Color(255, 255, 255), false)
+	// local pos2D = Get2DPosition(Lara:GetPosition())
+	// string:SetPosition(PercentToScreen(pos2D.x, pos2D.y))
+	static std::optional<Vec2> Get2DPosition(const Vec3& pos3D)
+	{
+		auto pos2D = g_Renderer.Get2DPosition(Vector3(pos3D.x, pos3D.y, pos3D.z));
```

Comment:
Maybe instead of reconstructing `Vector3` in place, make an operator which implicitly converts `Vec3` to `Vector3`?